lib/Test/WWW/Stub.pm

```diff
@@ -83,6 +83,18 @@ sub last_request {
     return $Requests[-1];
 }
 
+sub last_request_for {
+    my ($class, $method, $url) = @_;
+    my $reqs = { map { _request_signature($_) => $_ } @Requests };
+    my $signature = "$method $url";
+    return $reqs->{$signature};
+}
+
+sub _request_signature {
+    my ($req) = @_;
+    return join ' ', $req->method, $req->uri;
```

Comment:
In Test::WWW::Sub, query_string part of requests are ignored on dispatching.
And especially in case of `GET` requests, specifying exact url may be difficult.

Do you really need last request for EXACT matching url? If don't, maybe following interface is better, I guess.

```perl
$ua->get('http://HOST/foobar');
$ua->get('http://HOST/foo?q=1');
$ua->get('http://HOST/foo?foo=bar');

my $reqs = Test::WWW::Stub->requests_for_url('http://HOST/foo');
scalar @$reqs; # 2
$reqs->[-1]->uri; # http://HOST/foo?foo=bar
```